### ISSUE TITLE
[Fabric] Shim AccessibilityTraits to macOS

### ISSUE THREAD

**Issue by Saadnajmi** [microsoft/react-native-macos#2645]
Followup from #2293 :

Appkit doesn't have anything like `UIAccessibilityTraits`, which React Native uses. We currently do a lossy conversion of the "role" prop to `NSAccessibilityRole`, where in iOS, the role prop is used to set traits (And RCTAccessibilityTraits), which in turn also is used for things like `role` and `accessibilityValue`.

We could improve our accessibility story if we were to implement a shim for `UIAccessibilityTraits` in RCTUIKit.

**Comment by Saadnajmi** [comment]
We can maybe take inspiration from https://github.com/ChartsOrg/Charts/blob/master/Source/Charts/Utils/Platform+Accessibility.swift
